b2/sync.py

```diff
@@ -537,6 +537,12 @@ def _walk_relative_paths(self, prefix_len, dir_path, reporter):
         names = {}  # name to (full_path, relative path)
         dirs = set()  # subset of names that are directories
         for name in os.listdir(dir_path):
+            # We expect listdir() to return unicode if dir_path is unicode.
+            # This assertion is to help track down issue #190.  One theory
+            # is that listdir() is not behaving correctly.  If it were to
+            # return bytes rather than unicode, it's not clear how to decode it.
+            assert isinstance(name, six.text_type)
```

Comment:
I guess that for ASCII file names, the code will work even if the assertion fails.  It will decode bytes with the 'ascii' encoding, and come up with the right answer.

Maybe I'll just fail the assertion if there are non-ASCII characters in the file name, and the file name is not unicode.